src/Examine.Lucene/FacetExtensions.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Examine.Lucene.Search;
+using Lucene.Net.Facet.Range;
+
+namespace Examine.Lucene
+{
+    public static class FacetExtensions
+    {
+        /// <summary>
+        /// Get the values for a particular facet in the results
+        /// </summary>
+        public static IFacetResult GetFacet(this ISearchResults searchResults, string field)
+        {
+            if (!(searchResults is IFacetResults facetResults))
+            {
+                throw new ArgumentException("Result does not support facets");
+            }
+
+            facetResults.Facets.TryGetValue(field, out IFacetResult facet);
```

Comment:
what happens if this is false? This should be an if statement and if it's not found then probably throw an InvalidOperationException?

Reply:
With the support of Nullable in #313 this has been marked as returning the facet result or null. I think this is more developer friendly because you then don't risk throwing an exception when trying to get a value. Maybe a solution is to have a `GetFacet` which the exception and a `TryGetFacet` without? 
(https://github.com/Shazwazza/Examine/pull/313/files#diff-0e4473809aca1689b957462548caff26bf5e68cad97b062a031cff5a79488901)

You can also risk a facet value being null if there's no results in the query like in the test `Match_By_Path` where I've added this to the bottom of the test where no results are returned:

```csharp
var results3 = wildcardfilter.WithFacet("nodeName").Execute();
var facetResults3 = results3.GetFacet("nodeName");
Assert.AreEqual(0, results3.TotalItemCount);
Assert.AreEqual(0, facetResults3?.Count() ?? 0);
```
Here the addtion of an `InvalidOperationException` will throw and might not be expected by everyone.

Reply:
Ah right, I see, so GetFacet can return null, gotcha